batavia/types/List.js

```diff
@@ -72,14 +71,29 @@ batavia.types.List = function() {
     List.prototype.__lt__ = function(other) {
 
 
-        if(batavia.isinstance(other, [batavia.types.Bytes, batavia.types.Bytearray])){
+        if (batavia.isinstance(other, [batavia.types.Bytes, batavia.types.Bytearray])){
             throw new batavia.builtins.TypeError("unorderable types: list() < " + batavia.type_name(other) + "()")
         }
 
         if (other !== null) {
             if (batavia.isinstance(other, batavia.types.List)) {
-                /* update this line to get Pythonic list < list behavior */
-                return this.valueOf() < other;
+                // edge case where this==[]
+                if (this.length === 0 && other.length > 0){
+                    return true;
+                }
+
+                for (var i=0; i<this.length; i++){
+
+                    //other ran out of items.
+                    if (other[i] === undefined){
+                        return false;
+                    }
+                    if (this[i].__ne__(other[i])){
+                        return this[i].__lt__(other[i]);
+                    }
+                }
+                //got through loop and all values were equal. Determine by comparing length
+                return this.length < other.length;
```

Comment:
Oh! I was totally wrong, then. I thought two sequences had to be the same size to be compared. TIL!